### ISSUE TITLE
New Task: Automate ingestion of Lotek marine tags to ADR

### ISSUE THREAD

**Issue by laurabrenskelle** [ioos/marine_life_data_network#28]
### Who is requesting this?

@ioos/marine-life

### What is being requested?

Automate the ingestion of Lotek marine tags to the ATN Project Registration application (ADR). This includes PSATFLEX, PSATLIFE, PSATGEO, KiwiSat, and FastGPS.

### What is the requested deadline and why?

2024-07-01

### What is the current status quo (i.e., what happens if this does not get done)?

Ingestion of these tag styles may require extra work to be manually ingested and this will be more time consuming.

### What indicates this is done (i.e., how do we know this is complete)?

- [ ] The source code is published on GitHub at [ioos/ioos-atn-data](https://github.com/ioos/ioos-atn-data/) or another appropriate repository, and it is well documented.
- [ ] Data from these tags is being automatically ingested into ADR and the data are correct.

### Provide a description or any other important information.

cc @mmckinzie

**Comment by mmckinzie** [comment]
This task is on-hold for now. I have an upcoming meeting with Donna Kehoe from Lotek om Aug 14th to discuss feasibility of opening a direct pipeline to Lotek's data portals.

**Comment by mmckinzie** [comment]
I had a meeting with Lotek (Donna Kehoe) on Aug 15th to discuss. Lotek is updating their data portal, and access points and anticipates these updates to be complete by Spring 2026. Lotek, suggest we wait until their system updates are finalized before we resume effort to open an auto-ingest pipeline between Lotek and the ATN DAC. Follow-up meeting has been scheduled for January 22, 2026.

**Comment by MathewBiddle** [comment]
Sounds like this is pushed until Spring of 2026. Setting a start date of April 1 2026.

Moving to IceBox as the activity is unclear until we learn more from Lotek.

**Comment by mmckinzie** [comment]
@MathewBiddle Sorry, that was a typo on my part, I meant spring of 2025, not 2026. Although when I had my last check-in call with Lotek in early February 2025, they were a little behind schedule and expected updates wouldn't be finalized until later in 2025. I should have more information after attending the manufacturer's forum on April 30th, and my next check-in call is scheduled for early July. Those happen approximately every 6 months. 

Either way, fine to put activity on hold for now, let's revisit in the Fall 2025 (~ Sep 1) as opposed to Spring 2026.

**Comment by MathewBiddle** [comment]
Where does this sit? I like putting dev time into building automated data ingests. Can we make the code open source too?

@mmckinzie @laurabrenskelle

**Comment by mmckinzie** [comment]
The last I heard from Lotek, when I spoke with one of their reps at the NEPSS meeting a few weeks ago, was that they are still working on implementing internal changes to their data system. I recently emailed Donna, Director of Marine and Freshwater Systems, my usual POC at Lotek, to schedule a check-in call. I'd like to schedule the meeting before I go on leave on May 15; if not, we'll get something scheduled for early June.  I'll report back once I know more.

**Comment by mmckinzie** [comment]
Meeting scheduled with Lotek on Friday, June 5th.

**Comment by mmckinzie** [comment]
@MathewBiddle @laurabrenskelle Received a follow-up email from Lotek after our June 5th meeting. Donna Kehoe connected me with Peter Leijen, Engineering and Technical Product Manager. The new Lotek Cloud Service is now functional and includes improved API capabilities that the DAC can leverage to auto-ingest data from their web server. GPS and PSAT data are also now processed directly via the web server, rather than using a separate software package, thereby allowing us direct access to all required Argos and/or modeled data and data products. You can find documentation here: https://data.lotek.com/guidance. I am attempting to arrange for a meeting to discuss technical requirements and next steps for later in August. I've added Iwen Su (@iwensu0313) to the email for awareness and to coordinate a meeting time that works for all. Once we have a better idea of what it would take to connect to the LCS API and internal DAC requirements to integrate autoingested data files into the DAC pipelines, we can circle back on timing and feasibility of incorporating these activities into our workplan.

**Comment by MathewBiddle** [comment]
From what I'm seeing, that webservice is for displaying data as html web tables or maps, etc. Does this service actually provide a way for ATN to get data from lotek?

> Welcome to the Lotek Webserver, Lotek’s webservice for displaying various data generated by Lotek products

Thank you for coordinating requirements gathering. It will be good to know what resources are needed to make this happen.

**Comment by mmckinzie** [comment]
Yes, it does. At least the associated API does. I'm still fuzzy on the details, but yes, this will allow us direct access and the ability to auto-ingest data files. It seems like we'd need a unique API key and some info from researchers (e.g., tag id, serial number) to tap into the system to gain access to files and an accompanying JSON record with a recordType indicator to say what type of data is expected. Data would be available in near R/T on a deployment-by-deployment basis. Peter mentioned daily access, but I suspect we can scrape data at whatever frequency is required, e.g., every 6 hrs. But I'll know more after meeting later in August.

**Comment by mmckinzie** [comment]
The conversation about the autoingestion of Lotek data and the use of the new API is being further documented in issue #203 and is also related to issue #45.

**Comment by MathewBiddle** [comment]
Moving the content from #203 here. 

### Who is requesting this?

ATN

### What is being requested?

Currently Lotek manufactured tag data are ingested manually 
- PI uploads data to Research Workspace
- TT standardizes it and ingests it into the ATN DAC

Lotek has recently developed an API for external access: https://data.lotek.com/guidance

The request is for Tetra Tech ingest/development team to coordinate with the Lotek developer team and ATN Data Coordinator @mmckinzie to streamline the ingestion process for registered ATN data coming from Lotek-manufacturered tags. This involves
- [ ] Getting proper access to data that Tetra Tech has been granted access to by PIs (PIs can share data with specific users/accounts)
- [ ] Testing a data flow w/ Lotek and ATN Coordinator on one active project w/ data ready to be shared
- [ ] Finalize the new data collector for automatically pulling from Lotek API

### What is the requested deadline and why?

_No response_

### What is the current status quo (i.e., what happens if this does not get done)?

Data from Lotek-manufactured tags continue to be manually ingested, which is usually x2 effort of automated ingests where Tetra Tech can pull from an API or server directly. Lotek-manufacturered tags make up the second largest of ATN registered data

### What indicates this is done (i.e., how do we know this is complete)?

Data from Lotek can be automatically ingested via API instead of having to do manual wrangling on PI uploaded data.

### Provide a description or any other important information.

Lotek currently doesn't allow PIs to split a dataset corresponding to a unique Serial ID into unique deployments the way Wildlife Computers does, so Tetra Tech will need to leverage metadata provided in the ATN Data Registration app that PI fill out to identify unique deployments.

**Comment by iwensu0313** [comment]
Update - we have met and worked w/ Lotek to set up access to their API. Easiest if we test with real ATN data. Megan has been corresponding with some PIs/Projects that are willing to share the data with us. For one Megan is waiting to confirm IDs to share with us, the other is planning to be deployed mid Oct.

**Comment by MathewBiddle** [comment]
@mmckinzie moved to blocked until you get okay from PI to release Lotek data. Then we will evaluate what our priorities are and when to start work on this.

**Comment by MathewBiddle** [comment]
This build will not impact the NCEI WAF development.
